File: app/src/main/java/com/mufcryan/objectdetectiondemo/util/LogUtil.java
package com.mufcryan.objectdetectiondemo.util;

import android.util.Log;

public class LogUtil {

    public static boolean showLog = true;
    private static String TAG = "logutil---";


    public static void v(String msg) {
        v(TAG, msg);
    }

    public static void v(String tag, String msg) {
        if (showLog) {
            msg = formNull(msg);
            Log.v(tag, msg);
        }
    }

    public static void v(String tag, String msg, Throwable tr) {
        if (showLog) {
            msg = formNull(msg);
            Log.v(tag, msg, tr);
        }
    }

    public static void d(String msg) {
        d(TAG, msg);
    }

    public static void d(String tag, String msg) {
        if (showLog) {
            msg = formNull(msg);
            Log.d(tag, msg);
        }
    }

    public static void d(String tag, String msg, Throwable tr) {
        if (showLog) {
            msg = formNull(msg);
            Log.d(tag, msg, tr);
        }
    }

    public static void i(String msg) {
        i(TAG, msg);
    }

    public static void i(String tag, String msg) {
        if (showLog) {
            msg = formNull(msg);
            Log.i(tag, msg);
        }
    }

    public static void i(String tag, String msg, Throwable tr) {
        if (showLog) {
            msg = formNull(msg);
            Log.i(tag, msg, tr);
        }
    }

    public static void w(String msg) {
        w(TAG, msg);
    }

    public static void w(String tag, String msg) {
        if (showLog) {
            msg = formNull(msg);
            Log.w(tag, msg);
        }
    }

    public static void w(String tag, String msg, Throwable tr) {
        if (showLog) {
            msg = formNull(msg);
            Log.w(tag, msg, tr);
        }
    }

    public static void e(String msg) {
        e(TAG, msg);
    }

    public static void e(String tag, String msg) {
        if (showLog) {
            msg = formNull(msg);
            Log.e(tag, msg);
        }
    }

    public static void e(String tag, String msg, Throwable tr) {
        if (showLog) {
            msg = formNull(msg);
            Log.e(tag, msg, tr);
        }
    }

    private static String formNull(String value) {
        return value == null ? "null" : value;
    }

    public static void printeException(Throwable exception) {
        printeException(TAG, exception);
    }

    public static void printeException(String tag, Throwable exception) {
        if (showLog) {
            String msg = formNull(Log.getStackTraceString(exception));
            Log.w(tag, msg);
        }
    }
}

File: base/src/main/java/com/mufcryan/util/LogUtil.kt
package com.mufcryan.util

import android.util.Log

object LogUtil {
  var showLog = true
  private const val TAG = "logutil---"
  fun v(msg: String?) {
    v(TAG, msg)
  }

  fun v(tag: String?, msg: String?) {
    if (showLog) {
      Log.v(tag, msg ?: "")
    }
  }

  fun v(tag: String?, msg: String?, tr: Throwable?) {
    if (showLog) {
      Log.v(tag, msg, tr)
    }
  }

  fun d(msg: String?) {
    d(TAG, msg)
  }

  fun d(tag: String?, msg: String?) {
    if (showLog) {
      Log.d(tag, msg ?: "")
    }
  }

  fun d(tag: String?, msg: String?, tr: Throwable?) {
    if (showLog) {
      Log.d(tag, msg, tr)
    }
  }

  fun i(msg: String?) {
    i(TAG, msg)
  }

  fun i(tag: String?, msg: String?) {
    if (showLog) {
      Log.i(tag, msg ?: "")
    }
  }

  fun i(tag: String?, msg: String?, tr: Throwable?) {
    if (showLog) {
      Log.i(tag, msg, tr)
    }
  }

  fun w(msg: String?) {
    w(TAG, msg)
  }

  fun w(tag: String?, msg: String?) {
    if (showLog) {
      Log.w(tag, msg ?: "")
    }
  }

  fun w(tag: String?, msg: String?, tr: Throwable?) {
    if (showLog) {
      Log.w(tag, msg, tr)
    }
  }

  fun e(msg: String?) {
    e(TAG, msg)
  }

  fun e(tag: String?, msg: String?) {
    if (showLog) {
      Log.e(tag, msg ?: "")
    }
  }

  fun e(tag: String?, msg: String?, tr: Throwable?) {
    if (showLog) {
      Log.e(tag, msg, tr)
    }
  }

  private fun formNull(value: String?): String {
    return value ?: "null"
  }

  fun printException(exception: Throwable?) {
    printException(TAG, exception)
  }

  fun printException(tag: String?, exception: Throwable?) {
    if (showLog) {
      val msg = formNull(Log.getStackTraceString(exception))
      Log.w(tag, msg)
    }
  }
}